ouroboros-consensus/src/Ouroboros/Consensus/Mempool/TxSeq.hs

```diff
@@ -162,6 +168,25 @@ splitAfterTicketNo (TxSeq txs) n =
     case FingerTree.split (\m -> mMaxTicket m > n) txs of
       (l, r) -> (TxSeq l, TxSeq r)
 
+-- | \( O(\log(n) \). Look up a transaction in the sequence by its 'SlotNo'.
+--
+lookupBySlotNo :: TxSeq tx -> SlotNo -> Maybe tx
+lookupBySlotNo (TxSeq txs) n =
+    case FingerTree.search (\ml mr -> mMaxSlotNo ml >= n
+                                   && mMinSlotNo mr >  n) txs of
+      FingerTree.Position _ (TxTicket tx _ n') _ | n' == n -> Just tx
+      _                                                    -> Nothing
```

Comment:
Wait, there will be multiple transactions with the same `SlotNo`. During each slot, currently 20 seconds, many transactions can arrive that will all have the same `SlotNo`. I'm not sure (yet) why you need `lookupBySlotNo`?